ext/rbs_extension/main.c

```diff
@@ -1,12 +1,23 @@
 #include "rbs_extension.h"
+#include "rbs/util/rbs_constant_pool.h"
+
+#include "ruby/vm.h"
+
+static
+void Deinit_rbs_extension(ruby_vm_t *_) {
+  rbs_constant_pool_free(RBS_GLOBAL_CONSTANT_POOL);
+}
 
 void
 Init_rbs_extension(void)
 {
 #ifdef HAVE_RB_EXT_RACTOR_SAFE
   rb_ext_ractor_safe(true);
-#endif  
+#endif
   rbs__init_constants();
   rbs__init_location();
   rbs__init_parser();
+
+  rbs_constant_pool_init(RBS_GLOBAL_CONSTANT_POOL, 0);
+  ruby_vm_at_exit(Deinit_rbs_extension);
```

Comment:
I'm not very sure if we need the deinit function. How can a Ruby vm exit without process termination? (Maybe there’s a reason I’m not aware of…)

Reply:
~~Embedding CRuby?~~

Reply:
I just added this to minimize the amount of leaked memory at the time the process terminates, so I could more easily find other leaks with tooling like valgrind.

We can remove it and just rely on process termination to clean up everything, but that has the downside of adding noise to any memory-related tooling. Perhaps we could add a `DEBUG` flag, and only do the cleanup when it's true?

> How can a Ruby vm exit without process termination?

Not that I know of, but it seemed like the exact API for this (ther's also `atexit()`)

Reply:
> I could more easily find other leaks with tooling like valgrind.

It makes sense! Thanks for the explanation.